codes/utilities/gan_tournament_selection.py

```diff
@@ -0,0 +1,127 @@
+import numpy as np
```

Comment:
This evaluates and scores the refiners and discriminators using the tournament rating evaluation method with Glicko scoring.